### PR TITLE
(WIP) - Travis Setup

### PR DESCRIPTION
**Pull request by rshade**

### CODE REVIEW

**Comment by rshade** [comment]
@RyanJarv/ @martinb3  any ideas on the travis package errors, should I remove the tests

File: .kitchen.docker.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      - RUN yum -y install lsof which systemd-sysv initscripts wget net-tools
+
+- name: fedora-latest
+  driver:
+    image: fedora:latest
+    pid_one_command: /usr/lib/systemd/systemd
+    intermediate_instructions:
+      - RUN dnf -y install yum which systemd-sysv initscripts wget net-tools
+
+- name: ubuntu-12.04
```
Comment (tas50) [inline .kitchen.docker.yml]: 12.04 can get nuked at this point. It goes EOL next month

File: .kitchen.docker.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+- name: ubuntu-16.04
+  driver:
+    image: ubuntu:16.04
+    pid_one_command: /bin/systemd
+    intermediate_instructions:
+      - RUN /usr/bin/apt-get update
+      - RUN /usr/bin/apt-get install apt-transport-https lsb-release procps net-tools -y
+
+- name: opensuse-13.2
```
Comment (tas50) [inline .kitchen.docker.yml]: 13.2 went EOL at the beginning of the year

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  - /opt/chefdk/embedded/bin/cookstyle --version
+  - /opt/chefdk/bin/chef exec foodcritic --version
+
+script:
+ - KITCHEN_LOCAL_YAML=.kitchen.docker.yml /opt/chefdk/embedded/bin/kitchen verify ${INSTANCE}
+
+matrix:
+  include:
+    - script:
+      - /opt/chefdk/bin/chef exec rake
```
Comment (tas50) [inline .travis.yml]: You'll want to switch this over to delivery local all since you shipped the delivery config

File: Berksfile.lock [hunk 4]
```diff
@@ -20,14 +20,16 @@ GRAPH
   install_varnish (0.0.1)
     apt (>= 0.0.0)
     varnish (>= 0.0.0)
+    yum (>= 0.0.0)
```
Comment (tas50) [inline Berksfile.lock]: Is there a reason to ship the berkslock?

File: Gemfile [hunk 5]
```diff
@@ -1,22 +1,17 @@
 source 'https://rubygems.org'
 
 group :lint do
```
Comment (tas50) [inline Gemfile]: This whole file can probably go away. Everything minus kitchen-docker is in ChefDK and that's not being used here actually

File: recipes/configure.rb [hunk 6]
```diff
@@ -19,6 +19,8 @@
 #
 
 include_recipe 'apt'
+include_recipe 'yum'
```
Comment (tas50) [inline recipes/configure.rb]: yum itself probably isn't necessary. That doesn't prime cache like apt. It just configures the main yum config file

File: spec/unit/recipes/default_spec.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
+#
+# Cookbook:: filesystem
+# Spec:: default
+#
+# Copyright:: 2017, The Authors, All Rights Reserved.
```
Comment (tas50) [inline spec/unit/recipes/default_spec.rb]: A real copyright should get set here

File: Berksfile.lock (hunk 4 above)
Comment (RyanJarv) [inline Berksfile.lock]: Not as far as I know, think it makes sense to remove it.

File: test/fixtures/cookbooks/install_varnish/metadata.rb [hunk 8]
```diff
@@ -3,3 +3,5 @@
 version '0.0.1'
 depends 'varnish'
 depends 'apt'
+depends 'yum'
```
Comment (tas50) [inline test/fixtures/cookbooks/install_varnish/metadata.rb]: The main cookbook already depends on all of this so these can go away

File: Rakefile [hunk 9]
```diff
@@ -7,7 +7,7 @@ require 'kitchen'
 # Style tests. Rubocop and Foodcritic
 namespace :style do
   desc 'Run Ruby style checks'
-  RuboCop::RakeTask.new(:ruby)
+  sh('chef exec cookstyle -D --format offenses --display-cop-names')
```
Comment (tas50) [inline Rakefile]: Do we need the Rakefile at all now?

**Pull request merged**